File: 3270-find-the-key-of-the-numbers/3270-find-the-key-of-the-numbers.py
class Solution:
    def solve(self, n: int) -> str:
        return f"{n:04d}"
    
    def generateKey(self, num1: int, num2: int, num3: int) -> int:
        s1 = self.solve(num1)
        s2 = self.solve(num2)
        s3 = self.solve(num3)
        ans = 0
        
        for i in range(4):
            ans = ans * 10 + min(int(s1[i]), int(s2[i]), int(s3[i]))
        
        return ans

File: 3270-find-the-key-of-the-numbers/3270-find-the-key-of-the-numbers.cpp
class Solution {
public:
    string solve(int n) {
        if (n < 10) {
            return "000" + to_string(n);
        }
        if (n < 100) {
            return "00" + to_string(n);
        }
        if (n < 1000) {
            return "0" + to_string(n);
        }
        return to_string(n);
    }

    int generateKey(int num1, int num2, int num3) {
        string s1 = solve(num1);
        string s2 = solve(num2);
        string s3 = solve(num3);
        int ans = 0;
        for (int i = 0; i < 4; i++) {
            ans = ans * 10 + (min({s1[i], s2[i], s3[i]}) - '0');
        }
        return ans;
    }
};